.github/workflows/build-binaries-and-push-to-r2.yaml

```diff
@@ -0,0 +1,64 @@
+name: Upload core binary to Cloudflare R2 bucket
+
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+    inputs:
+      branch:
+        description: Github branch to checkout for build
+        required: true
+        default: main
+        type: string
+
+jobs:
+    build-and-upload:
+        name: Build binary and push to R2
+        runs-on: ubuntu-22.04
```

Comment:
we use ubuntu 20.04 in the other action. We should keep it consistent. We can bump it everywhere in another PR maybe (just need to let validators know since it's a breaking change, some libs need to be installed etc)
https://github.com/axelarnetwork/axelar-core/blob/main/.github/workflows/build-docker-image-and-binaries.yaml#L25

Reply:
shall we modify the existing build action to upload the binaries to cloudflare? This avoids having to run one more action for release